vscode-dotnet-runtime-library/src/Utils/FileUtilities.ts

```diff
@@ -0,0 +1,189 @@
+     */
+    public wipeDirectory(directoryToWipe : string)
+    {
+        if(!fs.existsSync(directoryToWipe))
+        {
+            return;
+        }
+
+        // Use rimraf to delete all of the items in a directory without the directory itself.
+        fs.readdirSync(directoryToWipe).forEach(f => fs.rmSync(`${directoryToWipe}/${f}`));
+    }
+
+    /**
+     *
+     * @param nodeArchitecture the architecture in node style string of what to install
+     * @returns the architecture in the style that .net / the .net install scripts expect
+     *
+     * Node - amd64 is documented as an option for install scripts but its no longer used.
+     * s390x is also no longer used.
+     * ppc64le is supported but this version of node has no distinction of the endianness of the process.
+     * It has no mapping to mips or other node architectures.
+     *
+     * @remarks Falls back to string 'auto' if a mapping does not exist which is not a valid architecture.
+     */
+    public nodeArchToDotnetArch(nodeArchitecture : string, eventStream : IEventStream)
+    {
+        switch(nodeArchitecture)
+        {
+            case 'x64': {
+                return nodeArchitecture;
+            }
+            case 'ia32': {
+                return 'x86';
+            }
+            case 'x86': {
+                // In case the function is called twice
+                return 'x86';
+            }
+            case 'arm': {
+                return nodeArchitecture;
+            }
+            case 'arm64': {
+                return nodeArchitecture;
+            }
+            case 's390x': {
+                return 's390x';
+            }
+            default: {
+                eventStream.post(new DotnetCommandFallbackArchitectureEvent(`The architecture ${os.arch()} of the platform is unexpected, falling back to auto-arch.`));
+                return 'auto';
+            }
+        }
+    }
+
+    /**
+     *
+     * @param nodeOS the OS in node style string of what to install
+     * @returns the OS in the style that .net / the .net install scripts expect
+     *
+     */
+    public nodeOSToDotnetOS(nodeOS : string, eventStream : IEventStream)
+    {
+        switch(nodeOS)
+        {
+            case 'win32': {
+                return 'win';
+            }
+            case 'darwin': {
+                return 'osx';
+            }
+            case 'linux': {
+                return nodeOS;
+            }
+            default: {
+                eventStream.post(new DotnetCommandFallbackOSEvent(`The OS ${os.platform()} of the platform is unexpected, falling back to auto-os.`));
+                return 'auto'
+            }
+        }
+    }
+
+    /**
+     *
+     * @returns true if the process is running with admin privileges
+     */
+    public isElevated() : boolean
+    {
+        if(os.platform() !== 'win32')
+        {
+            const commandResult = proc.spawnSync('id', ['-u']);
```

Comment:
for these commands, we should wrap them in a catch such that they dont report to the terminal upon failure (not sure if they are already or not)